### PR TITLE
llm: Add Anthropic support

### PR DESCRIPTION
**Pull request by Panchajanya1999**
The config goes like this:
{
	"llm.api_type": "anthropic",
        "llm.api_base": "https://api.anthropic.com/v1/messages",
        "llm.api_key": "sk-ant-xxxxx",
        "llm.model": "claude-3-opus"
}

(cherry picked from commit 0d95d3a3c831735d10865344dbfd8aa22173f214)
Signed-off-by: Panchajanya1999 <kernel@panchajanya.dev>

### CODE REVIEW

**Comment by Panchajanya1999** [comment]
@microsoft-github-policy-service agree

**Comment by Panchajanya1999** [comment]
hi @liqul can you have a look into this? It works perfectly.

**Comment by liqul** [comment]
> hi @liqul can you have a look into this? It works perfectly.

Thanks for your contribution. Could you add a markdown file [here](https://github.com/microsoft/TaskWeaver/tree/main/website/docs/llms) to explain how to configure it?

**Comment by liqul** [comment]
Could you address the test failure?

**Comment by Panchajanya1999** [comment]
> Could you address the test failure?

Yeah, just added a commit to fix it.

**Comment by Panchajanya1999** [comment]
Yeah, this is the doc.

File: taskweaver/llm/anthropic.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import os
+from typing import Any, Generator, List, Optional
+
+from anthropic import Anthropic
```
Comment (liqul) [inline taskweaver/llm/anthropic.py:4]: Move this line into __init__ function would fix the tests.

**Review by liqul — COMMENTED** [review]
It is not the way to fix the dependency issue. All LLMs must be optional, which means the user can choose not to install the libraries when they are using other LLM endpoints. Therefore, the way fixing the test issue is to move the `import` line into the __init__ of the endpoint service class. You can take a look at other implementations for reference.

File: requirements.txt [hunk 2]
```diff
@@ -24,6 +24,6 @@ pytest>=7.0.0
 vcrpy>=5.0.0
 colorama>=0.4.6
 
-
-
+# Anthropic
+anthropic>=0.30.0
```
Comment (liqul) [inline requirements.txt:28]: Should not add to the global requirements file.

**Comment by Panchajanya1999** [comment]
Hi @liqul can you check now?

**Review by liqul — APPROVED** [review]

**Pull request merged**